include/layers/Tensor.hpp

```diff
@@ -0,0 +1,166 @@
+#pragma once
+
+#include <cstdint>
+#include <iostream>
+#include <stdexcept>
+#include <type_traits>
+#include <vector>
+
+#include "layers/Shape.hpp"
+
+enum class Type { kUnknown, kInt, kDouble };
+
+template <typename T>
+std::vector<uint8_t>* to_byte(std::vector<T>& v) {
+  return reinterpret_cast<std::vector<uint8_t>*>(&v);
+}
+
+template <typename T>
+const std::vector<uint8_t>* to_byte(const std::vector<T>& v) {
+  return reinterpret_cast<const std::vector<uint8_t>*>(&v);
+}
+
+template <typename T>
+Type GetTypeEnum() {
+  if constexpr (std::is_same<T, int>::value) {
+    return Type::kInt;
+  } else if constexpr (std::is_same<T, double>::value) {
+    return Type::kDouble;
+  } else {
+    return Type::kUnknown;
+  }
+}
+
+class Tensor {
+ private:
+  Shape shape_;
+  std::vector<uint8_t> values_;
+  Type type_;
+
+  std::vector<uint8_t> SetRightTypeValues() {
+    if (type_ == Type::kInt) {
+      return std::vector<uint8_t>(shape_.count() * sizeof(int), 0);
+    }
+    if (type_ == Type::kDouble) {
```

Comment:
I think we need to use float instead double because double is redundant for cnn

Reply:
Fixed